deep_agent/src/infrastructure/tool_access.py

```
@@ -0,0 +1,219 @@
+    Returns:
+        A copy of the tool with its callable replaced by the approval wrapper.
+    """
+    payload = _build_hitl_payload(tool, agent_name)
+
+    coroutine = getattr(tool, "coroutine", None)
+    func = getattr(tool, "func", None)
+
+    if inspect.iscoroutinefunction(coroutine):
+        original_coroutine = coroutine
+
+        async def wrapped_coroutine(**kwargs: Any) -> Any:
```

Ruff (0.15.21)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `**kwargs`

(ANN401)